src/topic/TopicMessageQuery.js

```diff
@@ -332,6 +341,16 @@ export default class TopicMessageQuery {
                 (data) => {
                     const message = proto.ConsensusTopicResponse.decode(data);
 
+                    if (this._limit != null && this._limit.gt(0)) {
+                        this._limit = this._limit.sub(1);
+                    }
+
+                    this._startTime = Timestamp._fromProtobuf(
```

Comment:
Need to add one nanosecond

Reply:
Damn, good catch. Fixed this in https://github.com/hashgraph/hedera-sdk-js/pull/411